### ISSUE TITLE
Type Piracy, `Base.hcat`

### ISSUE THREAD

**Issue by kpa28-git** [JuliaDynamics/StateSpaceSets.jl#34]
**Describe the problem**

There is a type piracy with `Base.hcat` in the latest release. I don't know if this is intentional, but I decided to make an issue if it wasn't. I'm not loading StateSpaceSets.jl as a direct dependency.

**Minimal Working Example**
It looks like `hcat` with an `AbstractRange` argument calls the statespaceset.jl `Base.hcat` and errors:

`hcat(1:3, ones(3))`

> TypeError: in <:, expected Type, got a value of type Nothing

**Package versions**
* Julia v1.10.5
* StateSpaceSets.jl v2.1.0

**Comment by Datseris** [comment]
oh shit. Thanks a lot for opening this. Not intendend. It's a bug from releasing 2.0. Can you please provide some actual code that leads to the issue? `hcat` needs two arguments.

**Comment by kpa28-git** [comment]
Good, hope this was useful at least in a small way.

Base.hcat of two Vectors  does not fail (though I'm guessing there's still a type piracy). As you can see, hcat with an `<:AbstractRange` has type piracy and fails:
```julia
julia> using StateSpaceSets

julia> hcat(collect(1:3), ones(3))
3×2 Matrix{Float64}:
 1.0  1.0
 2.0  1.0
 3.0  1.0

julia> hcat(1:3, ones(3))
ERROR: TypeError: in <:, expected Type, got a value of type Nothing
Stacktrace:
 [1] hcat(::UnitRange{Int64}, ::Vector{Float64})
   @ StateSpaceSets ~/.julia/packages/StateSpaceSets/ZtgR9/src/statespaceset.jl:102
 [2] top-level scope
   @ REPL[3]:1
   ```

**Comment by Datseris** [comment]
Thank you so much for reporting this. This was a critical bug. Fixed now in v2.1.1.
